### DISCUSSION TITLE
Can I use any AI model/service with markitdown?

### DISCUSSION THREAD

**Discussion by brocksshoppe** [microsoft/markitdown#174] [Q&A]
I see that OpenAI was used in the code example but I was wondering, does markitdown work best with OpenAI or can I use my own model or something else? Just wondering if there are any restrictions here.

**Comment by Batman313v** [comment]
Havn't tested it yet but in theory you should be able to replace the `client` with any OpenAI compatible API. Example for ollama:
```py
from markitdown import MarkItDown
from openai import OpenAI

client = OpenAI(
	base_url='http://ollamahost:11434/v1/',
)
md = MarkItDown(llm_client=client, llm_model="llama3.2")
result = md.convert("example.jpg")
print(result.text_content)
```
I am going to message around with using the ollama api directly at some point but for now this should work.

I just found this repo so I havn't had time to really get into it but if they use tools or function calling (probably do) then you need to make sure your api provider and model support it.

**Reply by dasgoswami** [reply]
does it work?
